### ISSUE TITLE
hdr.table with fields 

### ISSUE THREAD

**Issue by licode** [bluesky/databroker#391]
In the current implementation of databroker, hdr.table or hdr.events function with fill=True and fields=[a,b,c] works as the following:

1. First fill all the data if their type is external
2. Then select data_key in [a,b,c] and output

It is better to first select data in the given fields, then only fill those selected data and output. This will make loading data much faster and efficient.

**Comment by danielballan** [comment]
That is true. If I proceed as planned, I will gut the entire databroker codebase by the end of the next week (replacing it with intake) so I suggest we not bother optimizing the current implementation.
